### PR TITLE
[Dotnet] Implement EKS Linux E2E Test

### PR DESCRIPTION
**Pull request by XinRanZhAWS**
*Description of changes:*
Implement EKS Linux for Dotnet E2E Canary
*Rollback procedure:*
Revert it
<Can we safely revert this commit if needed? If not, detail what must be done to safely revert and why it is needed.>
Yes
*Ensure you've run the following tests on your changes and include the link below:*
*Test*
https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/10742366614

To do so, create a `test.yml` file with `name: Test` and workflow description to test your changes, then remove the file for your PR. Link your test run in your PR description. This process is a short term solution while we work on creating a staging environment for testing.

NOTE: TESTS RUNNING ON A SINGLE EKS CLUSTER CANNOT BE RUN IN PARALLEL. See the [needs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds) keyword to run tests in succession.
- Run Java EKS on `e2e-playground` in us-east-1 and eu-central-2
- Run Python EKS on `e2e-playground` in us-east-1 and eu-central-2
- Run metric limiter on EKS cluster `e2e-playground` in us-east-1 and eu-central-2
- Run EC2 tests in all regions
- Run K8s on a separate K8s cluster (check IAD test account for master node endpoints; these will change as we create and destroy clusters for OS patching)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: validator/src/main/resources/expected-data-template/dotnet/eks/linux/aws-sdk-call-trace.mustache [hunk 1]
```diff
@@ -0,0 +1,68 @@
+[{
+  "name": "^{{serviceName}}$",
+  "http": {
+    "request": {
+      "url": "^{{endpoint}}/aws-sdk-call$",
+      "method": "^GET$"
+    },
+    "response": {
+      "status": "^500$"
+    }
```
Comment (majanjua-amzn) [inline validator/src/main/resources/expected-data-template/dotnet/eks/linux/aws-sdk-call-trace.mustache:10]: Remove all status checks, not relevant. We will follow up with other canaries for this change too

File: .github/workflows/dotnet-eks-canary.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  group: ${{ github.workflow }}
+  cancel-in-progress: false
+
+permissions:
+  id-token: write
+  contents: read
+
+
+jobs:
+  dotnet-e2e-eks-test:
```
[the remark this answers is not in the document]
Comment (XinRanZhAWS) [reply .github/workflows/dotnet-eks-canary.yml:27]: Done

File: .github/workflows/dotnet-eks-test.yml [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+
+      - name: Add eksctl to Github Path
+        run: |
+          echo "${{ github.workspace }}/eksctl" >> $GITHUB_PATH
+
+      # This step deletes lingering resources from previous test runs
+      - name: Delete all sample app namespaces
+        continue-on-error: true
+        timeout-minutes: 5
+        run: kubectl get namespace | awk '/^ns-[0-9]+-[0-9]+/{print $1}' | xargs kubectl delete namespace
```
[the remark this answers is not in the document]
Comment (XinRanZhAWS) [reply .github/workflows/dotnet-eks-test.yml:142]: done

File: validator/src/main/resources/expected-data-template/dotnet/eks/linux/aws-sdk-call-trace.mustache (hunk 1 above)
Comment (XinRanZhAWS) [reply validator/src/main/resources/expected-data-template/dotnet/eks/linux/aws-sdk-call-trace.mustache]: Done

File: .github/workflows/dotnet-eks-test.yml [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+        with:
+          command: "cd ${{ env.TEST_RESOURCES_FOLDER }}/terraform/dotnet/eks/linux && terraform init && terraform validate"
+          cleanup: "rm -rf .terraform && rm -rf .terraform.lock.hcl"
+          max_retry: 6
+          sleep_time: 60
+
+      - name: Set Sample App Image
+        run: |
+          echo MAIN_SAMPLE_APP_IMAGE_ARN="${{ env.ACCOUNT_ID }}.dkr.ecr.${{ env.E2E_TEST_AWS_REGION }}.amazonaws.com/${{ secrets.APP_SIGNALS_DOTNET_E2E_FE_SA_IMG }}" >> $GITHUB_ENV
+          echo REMOTE_SAMPLE_APP_IMAGE_ARN="${{ env.ACCOUNT_ID }}.dkr.ecr.${{ env.E2E_TEST_AWS_REGION }}.amazonaws.com/${{ secrets.APP_SIGNALS_DOTNET_E2E_RE_SA_IMG }}" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (XinRanZhAWS) [reply .github/workflows/dotnet-eks-test.yml]: Done

**Review by vastin — APPROVED** [review]

**Pull request merged**